src/mkanki.js

```diff
@@ -1,149 +1,106 @@
 
-const cssData = fs.readFileSync(HTML_DIR + DEFAULT_CSS_FILENAME, 'utf8');
+const cssData = fs.readFileSync(HTML_DIR + DEFAULT_CSS_FILENAME, "utf8");
 
 const model = new anki.Model({
   name: DEFAULT_NOTE_TYPE_NAME,
   id: Date.now().toString(),
-  flds: [{ name: 'Word' }, { name: 'Front' }, { name: 'Back' }],
-  req: [[0, 'all', [0]]],
+  flds: [{ name: "Word" }, { name: "Front" }, { name: "Back" }],
+  req: [[0, "all", [0]]],
   tmpls: [
     {
-      name: 'Card 1',
-      qfmt: '{{Front}}',
-      afmt: '{{FrontSide}}\n\n<hr id=answer>\n\n{{Back}}',
+      name: "Card 1",
+      qfmt: "{{Front}}",
+      afmt: "{{FrontSide}}\n\n<hr id=answer>\n\n{{Back}}",
     },
   ],
   css: cssData,
 });
 const deck = new anki.Deck(Date.now(), DEFAULT_DECK_NAME);
 const package = new anki.Package();
 
-async function processHtmlFiles() {
-  fs.readdir(HTML_DIR, async (err, files) => {
-    if (err) {
-      writeToLog(err.message);
-      return;
-    }
-
-    async function addNote(file) {
-      if (path.extname(file) === '.html') {
-        const filePath = path.join(HTML_DIR, file);
-        fs.readFile(filePath, 'utf8', (err, data) => {
-          if (err) {
-            writeToLog(err.message);
-            return;
-          }
-          try {
-            const $ = cheerio.load(data);
-            let front = '';
-            $('.pos-header.dpos-h')
-              .toArray()
-              .forEach((ele) => {
-                front += $(ele).html() + "<hr color='grey'>";
-              });
-            const back = $('body').html();
-            const tags = $('.def-info.ddef-info')
-              .toArray()
-              .map((ele) => $(ele).find("span[class*='epp-xref dxref']").html())
-              .filter((ele) => ele !== null);
-            let tagResult = null;
-            if (tags.length != 0) {
-              const tagsDictinct = [...new Set(tags)].sort();
-              tagResult = tagsDictinct;
+async function processHtmlFilesNew() {
+  try {
+    const files = await new Promise((resolve, reject) => {
+      fs.readdir(HTML_DIR, (err, files) => {
+        if (err) {
+          writeToLog(err.message);
+          reject(err);
+        }
+        resolve(files);
+      });
+    });
+    let cnt = 0;
+    function addNote(file) {
+      return new Promise((resolve, reject) => {
+        if (path.extname(file) === ".html") {
+          const filePath = path.join(HTML_DIR, file);
+          fs.readFile(filePath, "utf8", (err, data) => {
+            if (err) {
+              writeToLog(err.message);
+              reject(err);
+            } else {
+              try {
+                const $ = cheerio.load(data);
+                let front = "";
+                $(".pos-header.dpos-h")
+                  .toArray()
+                  .forEach((ele) => {
+                    front += $(ele).html() + "<hr color='grey'>";
+                  });
+                const back = $("body").html();
```

Comment:
Some HTML file could be empty, we could throw a warn:

```js
                if (!front || !back) {
                  writeToLog(`[WARN] The file ${file} is empty`);
                }
```

Or just `reject()` the blank file's generation. I prefer reject it, cause generating a blank card seems meaningless.